test/integration.test.ts

```diff
@@ -185,23 +187,12 @@ describe('Starting cardano-wallet (and its node)', () => {
       };
     });
 
-    await launcher
-      .start()
-      .then(api => {
-        console.error('promise succeeded', api);
-      })
-      .catch(e => {
-        console.log('caught launcher exception', e);
-        expect(e.message).toEqual(
-          [
-            'cardano-wallet-jormungandr exited with status 0',
-            'jormungandr exited with status 1',
-          ].join('\n')
-        );
-      })
-      .finally(() => {
-        cleanupTestLauncher(launcher);
-        expect.assertions(1);
-      });
+    await expect(launcher.start().finally(() => cleanupTestLauncher(launcher)))
+      .rejects.toThrow(
+        [
+          'cardano-wallet-jormungandr exited with status 0',
+          'jormungandr exited with status 1',
+        ].join('\n')
+      );
```

Comment:
This doesn't work because `expect(…).rejects.toThrow` doesn't wrap around the promise invocation. Basically what happens is that you kick-off the promise by `launcher.start()` (in the context of the test case, not the expect function). By the time that you are calling `.rejects.toThrow` the promise might already have been thrown.

So the correct way would be to wrap `launcher.start()` into a separate function like this:
```js
    function startLauncher() {
      launcher.start().finally(() => cleanupTestLauncher(launcher))
    }
    await expect(startLauncher).rejects.toThrow(
        [
          'cardano-wallet-jormungandr exited with status 0',
          'jormungandr exited with status 1',
        ].join('\n')
      );
```

I haven't tested it in your case but that's the general idea 😉 hope it helps 🙏

Reply:
According to the [docs](https://jestjs.io/docs/en/tutorial-async#asyncawait), it should work as-is

```javascript
// async/await can be used.
it('works with async/await', async () => {
  expect.assertions(1);
  const data = await user.getUserName(4);
  expect(data).toEqual('Mark');
});

// async/await can also be used with `.resolves`.
it('works with async/await and resolves', async () => {
  expect.assertions(1);
  await expect(user.getUserName(5)).resolves.toEqual('Paul');
});
```
Verified in https://github.com/input-output-hk/cardano-launcher/pull/60/files#diff-a747c82eb3f3663f720d7af32aa57675R190-R196 which has a passing status in all envs